### PR TITLE
[12.0][FIX] product_variant_default_code: fix product.template write to accept multiple recordset

### PR DESCRIPTION
**Pull request by ernestotejeda**
Fix product.template write to accept multiple recordset

Cc @Tecnativa TT21023

### CODE REVIEW

File: product_variant_default_code/models/product.py [hunk 1]
```diff
@@ -123,22 +123,26 @@ def create(self, vals):
...
+        mask_templates = self.env['product.template']
         product_obj = self.env['product.product']
         if ('reference_mask' in vals and not vals['reference_mask'] or not
                 self.user_has_groups(
                     'product_variant_default_code.group_product_default_code'
                 )):
-            if self.attribute_line_ids:
-                vals['reference_mask'] = self._get_default_mask()
-        result = super(ProductTemplate, self).write(vals)
+            for template in self.filtered("attribute_line_ids"):
```
Comment (pedrobaeza) [inline product_variant_default_code/models/product.py]: No need of keeping another recordset

```suggestion
            with_variants = self.filtered("attribute_line_ids")
            for template in with_variants:
                ...
        super(ProductTemplate, self - with_variants).write(vals)
```

Comment (ernestotejeda) [reply product_variant_default_code/models/product.py]: Right, thanks @pedrobaeza , changes done

**Comment by ernestotejeda** [comment]
@pedrobaeza , Changes done, ready to be reviewed

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Let's fast-track it as obvious problem

/ocabot merge patch

**Pull request merged**